packages/muelu/src/Utils/MueLu_AlgebraicPermutationStrategy_def.hpp

```diff
@@ -116,7 +118,7 @@ namespace MueLu {
 
       if(maxVal > Teuchos::ScalarTraits<MT>::zero ()) { // keep only max Entries \neq 0.0
         permutedDiagCandidates.push_back(std::make_pair(grow,gMaxValIdx));
-        Weights.push_back(maxVal/(norm1*Teuchos::ScalarTraits<Scalar>::magnitude(nnz)));
+        Weights.push_back(maxVal/(norm1*Teuchos::as<MT>(nnz)));
       } else {
         std::cout << "ATTENTION: row " << grow << " has only zero entries -> singular matrix!" << std::endl;
```

Comment:
For a future commit, please consider not printing to stdout on every process -- that could be really bad at scale.  I think you have a way to print warnings, right?